File: rb/features.rb
# QuoteRetrieval SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module QuoteRetrievalFeatures
  def self.make_feature(name)
    case name
    when "base"
      QuoteRetrievalBaseFeature.new
    when "test"
      QuoteRetrievalTestFeature.new
    else
      QuoteRetrievalBaseFeature.new
    end
  end
end

File: py/features.py
# QuoteRetrieval SDK feature factory

from feature.base_feature import QuoteRetrievalBaseFeature
from feature.test_feature import QuoteRetrievalTestFeature


def _make_feature(name):
    features = {
        "base": lambda: QuoteRetrievalBaseFeature(),
        "test": lambda: QuoteRetrievalTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
